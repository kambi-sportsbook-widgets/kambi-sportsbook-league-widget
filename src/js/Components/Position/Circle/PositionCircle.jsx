import React from 'react';
import PropTypes from 'prop-types';
import styles from './PositionCircle.scss';

const COLORS = {
   GREEN: {
      background: 'KambiWidget-winPrimaryColor dark',
      text: 'KambiWidget-winSecondaryColor',
   },
   LIGHT_GREEN: {
      background: 'KambiWidget-winPrimaryColor light',
      text: 'KambiWidget-winSecondaryColor',
   },
   ORANGE: {
      background: 'KambiWidget-drawPrimaryColor dark',
      text: 'KambiWidget-drawSecondaryColor',
   },
   LIGHT_ORANGE: {
      background: 'KambiWidget-drawPrimaryColor light',
      text: 'KambiWidget-drawSecondaryColor',
   },
   RED: {
      background: 'KambiWidget-losePrimaryColor dark',
      text: 'KambiWidget-loseSecondaryColor',
   },
   LIGHT_RED: {
      background: 'KambiWidget-losePrimaryColor light',
      text: 'KambiWidget-loseSecondaryColor',
   },
   TRANSPARENT: {
      background: 'transparent',
      text: 'KambiWidget-card-text-color',
   },
};

const PositionCircle = ({ color, children }) => {
   let backgroundCss = `${styles.wrapper}`;
   let textCss = '';
   if (color == null) {
      color = 'TRANSPARENT';
   }
   backgroundCss += ` ${COLORS[color].background}`;
   textCss += ` ${COLORS[color].text}`;

   return (
      <span className={backgroundCss}>
         <span className={textCss}>
            {children}
         </span>
      </span>
   );
}

PositionCircle.propTypes = {

   /**
    * Current participant position
    */
   color: PropTypes.oneOf(Object.keys(COLORS)),

   /**
    * Optional node te be displayed inside circle
    */
   children: PropTypes.node

};

PositionCircle.defaultProps = {
   color: null
};

PositionCircle.COLORS = COLORS;

export default PositionCircle;
