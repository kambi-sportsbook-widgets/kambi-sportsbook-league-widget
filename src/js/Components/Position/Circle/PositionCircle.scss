:local(.wrapper) {
   border-radius: 50%;
   width: 22px;
   height: 22px;
   display: inline-block;
   background: #fff;
   line-height: 0;

   &.no-circle {
      background: transparent;
   }
}

:local(.circle) {
   border-radius: 50%;
   width: 22px;
   height: 22px;
   display: inline-block;
   line-height: 21px;
   text-align: center;
   font-size: 12px;

   &.light {
      opacity: .6;
   }

   &.transparent {
      background: transparent;
   }
}
