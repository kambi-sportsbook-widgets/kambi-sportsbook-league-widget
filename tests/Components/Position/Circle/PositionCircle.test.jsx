/* eslint-env jest */
import React, { Children } from 'react';
import PositionCircle from '../../../../src/js/Components/Position/Circle/PositionCircle';
import ReactTestRenderer from 'react-test-renderer';

describe('PositionCircle DOM rendering', () => {

   it('renders correctly with default props', () => {
      const tree = ReactTestRenderer.create(
         <PositionCircle/>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with given color', () => {
      const tree = ReactTestRenderer.create(
         <PositionCircle color={PositionCircle.COLORS.GREEN} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with transparent color', () => {
      const tree = ReactTestRenderer.create(
         <PositionCircle color={PositionCircle.COLORS.TRANSPARENT} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with given child element', () => {
      const tree = ReactTestRenderer.create(
         <PositionCircle>t</PositionCircle>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

});
