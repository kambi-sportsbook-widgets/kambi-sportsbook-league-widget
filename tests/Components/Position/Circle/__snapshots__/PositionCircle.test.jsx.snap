// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionCircle DOM rendering renders correctly with default props 1`] = `
<span
  className="wrapper transparent"
>
  <span
    className=" KambiWidget-card-text-color"
  />
</span>
`;

exports[`PositionCircle DOM rendering renders correctly with given child element 1`] = `
<span
  className="wrapper transparent"
>
  <span
    className=" KambiWidget-card-text-color"
  >
    t
  </span>
</span>
`;

exports[`PositionCircle DOM rendering renders correctly with given color 1`] = `
<span
  className="wrapper KambiWidget-winPrimaryColor dark"
>
  <span
    className=" KambiWidget-winSecondaryColor"
  />
</span>
`;

exports[`PositionCircle DOM rendering renders correctly with transparent color 1`] = `
<span
  className="wrapper transparent"
>
  <span
    className=" KambiWidget-card-text-color"
  />
</span>
`;
